### PR TITLE
Surface SDK version errors more cleanly

### PR DESCRIPTION
**Pull request by MarkWesleyAtAnki**
SDKVersionMismatch can now be caught by client code, if they wish to handle it separately.
SDKVersionMismatch exception now includes SDK version along with required and actual App versions.
After detecting the version mismatch, any remaining messages on the connection are ignored (to avoid throwing exceptions when they fail to unpack)
If no valid device connection is found, and an SDKVersionMismatch was detected on any device, then that SDKVersionMismatch exception is returned instead of a general "no devices found" exception.

### CODE REVIEW

File: src/cozmo/run.py [hunk 1]
```diff
@@ -276,6 +277,8 @@ def _remove_forward(self, serial):
                     try:
                         await conn_check(proto)
                     except Exception as e:
+                        if isinstance(e, exceptions.SDKVersionMismatch):
```
Comment (shawnblakesley) [inline src/cozmo/run.py]: Might be better to have a separate `except exceptions.SDKVersionMismatch as e:` branch than to use `isinstance` here

File: src/cozmo/run.py [hunk 2]
```diff
@@ -286,6 +289,10 @@ def _remove_forward(self, serial):
             except:
```
Comment (shawnblakesley) [inline src/cozmo/run.py]: Using an `except` branch here for `SDKVersionMismatch` which then `raise`s that error might be cleaner than using the `version_mismatch` variable

File: src/cozmo/run.py (hunk 1 above)
Comment (MarkWesleyAtAnki) [reply src/cozmo/run.py]: I did originally do it that way, but I felt it was duplicating more code that way. I'm not hugely torn either way - if there's a best-of-both-worlds I'd be interested?

File: src/cozmo/run.py (hunk 2 above)
Comment (MarkWesleyAtAnki) [reply src/cozmo/run.py]: I want it to try each Android in turn, and only raise SDKVersionMismatch if they all fail

**Pull request merged**